### DISCUSSION TITLE
Relacionamento

### DISCUSSION THREAD

**Discussion by RaMadaSilva** [balta-io/forum#134] [Tira Dúvidas] [answered]
Bom dia Mestre. 
Estou na carreira Back-end, no curso de fundamentos de AspNet 6, no modulo de Autenticação e Autorização. 

Ali o professor apresentou os dados já carregados nas tabelas e seus relacionamentos. 

eu queria saber como posso fazer esse preenchimento das reações a partir da API. 

eu fiz esse preenchimento a partir de uma console aplication, mas não tenho certeza se a forma como eu fiz é a forma mais correcta. 

Ex: 
![image](https://user-images.githubusercontent.com/91338367/204756599-bb1f15a1-f315-438d-92b7-e2ced4a48572.png)
Eu listo todos os usuarios; e peço para escolher um id, e listo todos os perfis e peço para escolher um id do perfil e salvo na tabela do relacionamento. 
 Aqui  vai o meu método para o vinculo. 

 public void Vincular()
        {
            var repoUser = new Repository<User>(DataBase.connection);
            var repoRole = new Repository<Role>(DataBase.connection);

            var users = repoUser.Get();
            var roles = repoRole.Get();

            Console.Clear();
            Console.WriteLine("Usuario");
            foreach (var user in users)
                Console.WriteLine($"{user.Id} - {user.Name}");
            Console.Write("Escolha um Id do usario: ");
            var idUser = int.Parse(Console.ReadLine());

            Console.WriteLine();
            Console.WriteLine("Perfis");
            foreach (var role in roles)
                Console.WriteLine($"{role.Id} - {role.Name}");
            Console.Write("Escolha um Id do Perfil: ");
            var idRole = int.Parse(Console.ReadLine());

            var userRole = new UserRole()
            {
                UserId = idUser,
                RoleId = idRole

            };

            var vincular = new Repository<UserRole>(DataBase.connection);
            vincular.Create(userRole);

            Program.Menu();

        }

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @RaMadaSilva, como vai?

Ótima pergunta... você pode fazer o cadastro assim:
```csharp
public Task<IActionResult> CreateAsync([FromBody]Role role)
{
    // Validações
    // Try/Catch 

    await context.Roles.AddAsync(role);
    await context.SaveChangesAsync();
    // Retorno para tela
}
```

O mesmo vale para o usuário... todos os métodos são iguais os que ensino no curso... não mostrei nos vídeos para não ficar repetitivo.
